server/src/main/java/org/candlepin/model/Content.java

```diff
@@ -148,6 +172,103 @@ public Content(String id) {
+     * <p/>
+     * Unlike the merge method, all properties from the source content are copied, including the
+     * state of any null collections and any identifier fields.
+     *
+     * @param source
+     *  The Content instance to copy
+     */
+    protected Content(Content source) {
+        this.setUuid(source.getUuid());
+        this.setId(source.getId());
+
+        this.setCreated(source.getCreated());
+        this.setUpdated(source.getUpdated());
+
+        this.setType(source.getType());
+        this.setLabel(source.getLabel());
+        this.setOwners(source.getOwners());
+        this.setName(source.getName());
+        this.setVendor(source.getVendor());
+        this.setContentUrl(source.getContentUrl());
+        this.setRequiredTags(source.getRequiredTags());
+        this.setReleaseVer(source.getReleaseVer());
+        this.setGpgUrl(source.getGpgUrl());
+        this.setMetadataExpire(source.getMetadataExpire());
+        this.setModifiedProductIds(source.getModifiedProductIds());
+        this.setArches(source.getArches());
+    }
+
+    /**
+     * Copies several properties from the given content on to this content instance. Properties that
+     * are not copied over include any identifiying fields (UUID, ID), the creation date and locking
+     * states. Values on the source content which are null will be ignored.
+     *
+     * @param source
+     *  The source content instance from which to pull content information
+     *
+     * @return
+     *  this content instance
+     */
+    public Content merge(Content source) {
+        this.setUpdated(source.getUpdated());
+
+        if (source.getType() != null) {
+            this.setType(source.getType());
+        }
+
+        if (source.getLabel() != null) {
+            this.setLabel(source.getLabel());
+        }
+
+        if (source.getName() != null) {
+            this.setName(source.getName());
+        }
+
+        if (source.getVendor() != null) {
+            this.setVendor(source.getVendor());
+        }
+
+        if (source.getContentUrl() != null) {
+            this.setContentUrl(source.getContentUrl());
+        }
+
+        if (source.getRequiredTags() != null) {
+            this.setRequiredTags(source.getRequiredTags());
+        }
+
+        if (source.getReleaseVer() != null) {
+            this.setReleaseVer(source.getReleaseVer());
+        }
+
+        if (source.getGpgUrl() != null) {
+            this.setGpgUrl(source.getGpgUrl());
+        }
+
+        if (source.getMetadataExpire() != null) {
+            this.setMetadataExpire(source.getMetadataExpire());
+        }
+
+        if (source.getModifiedProductIds() != null) {
+            this.setModifiedProductIds(source.getModifiedProductIds());
+        }
+
+        if (source.getArches() != null) {
+            this.setArches(source.getArches());
+        }
+
+        return this;
+    }
+
+    @Override
+    public Object clone() {
+        return new Content(this);
```

Comment:
The part I don't like about either of those options directly is it violates the dependency injection design we're attempting to adhere to. Clone is nice since you don't need to know what specific instance you're cloning to get a valid clone, whereas with the static factory method and the copy constructor, you have to know, at design time, what instance you'll be working with.

I know in our case that's currently not a problem, but I'd rather avoid the problem the first go around instead of planting landmines for others.

I'll update the clone method to behave properly.